### PR TITLE
removing Windows CRLF char

### PR DESCRIPTION
**Pull request by Tianhao-Gu**
Windows newline char CRLF fails FASTQ validator. 

https://kbase-jira.atlassian.net/browse/PTV-1377

### CODE REVIEW

**Comment by MrCreosote** [comment]
This seems like a weird thing to have to fix manually. I'd assume there's some way to get python to automatically convert windows line feeds to unix - have you looked into that?

**Comment by MrCreosote** [comment]
Might be useful: https://stackoverflow.com/a/2717154/643675

**Comment by MrCreosote** [comment]
maybe just ditch the check then fix optimization and just always run it through the fix loop with the python line converter

**Comment by Tianhao-Gu** [comment]
@MrCreosote Yea. Python `open(file_path)` can handle it just fine. But it's the 3rd party validator that throws an error:
https://github.com/kbaseapps/ReadsUtils/blob/master/lib/ReadsUtils/ReadsUtilsImpl.py#L895-L897

**Comment by MrCreosote** [comment]
that's why I'm saying to always run the fix loop, so you always remove blank lines and windows LF chars. Doing that is probably a whole lot faster than downloading the file so I doubt it'll add much time to the overall process

**Comment by MrCreosote** [comment]
Should probably file a bug with the 3rd party dev too

**Comment by Tianhao-Gu** [comment]
@MrCreosote sure. Just update code to always run fix loop.

File: kbase.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ service-language:
     python
 
 module-version:
-    0.3.12
+    0.3.13
 
 owners:
     [gaprice, msneddon, jkbaumohl]
```
Comment (MrCreosote) [inline kbase.yml]: Add yourself

**Review by MrCreosote — CHANGES_REQUESTED** [review]
LGTM other than 1 thing

File: lib/ReadsUtils/ReadsUtilsImpl.py [hunk 2]
```diff
@@ -862,28 +862,19 @@ def validateFASTQ(self, ctx, params):
...
-                        l = l.strip()
-                        if l:
-                            t.write(l + '\n')
-                            c += 1
-                    s.close()
-                    t.flush()
-                    shutil.copy2(t.name, file_path)
+            self.log('Removing blank lines and CRLF characters if any')
+            with open(file_path) as s, tempfile.NamedTemporaryFile(
+                    mode='w', dir=self.scratch) as t:
```
Comment (sean-mccorkle) [inline lib/ReadsUtils/ReadsUtilsImpl.py:869]: I think this is a big improvement in the code

**Review by sean-mccorkle — COMMENTED** [review]
This is not a show stopper, but if it were me, I would rename the function ValidateFASTQ to ValidateAndCleanFASTQ since its doing some cleanup on the file as well.

Also (again) not a show stopper:   I know FASTA files were not included in this, and I don't know of any occurrences of ^M's in FASTA files being reported, but is it worth considering checking those as well for this condition?  Or should that wait for an actual occurrence to be reported?

**Comment by MrCreosote** [comment]
> This is not a show stopper, but if it were me, I would rename the function ValidateFASTQ to ValidateAndCleanFASTQ since its doing some cleanup on the file as well.

It's not really though - it's making a temporary file with an arbitrary name and making the changes there and discarding it when done. It doesn't fix the original file or provide the path to the new file AFIAK.

**Comment by MrCreosote** [comment]
> This is not a show stopper, but if it were me, I would rename the function ValidateFASTQ to ValidateAndCleanFASTQ

That changes the API, not worth it IMO. Updating the docs would be good.

**Comment by sean-mccorkle** [comment]
> That changes the API, not worth it IMO. Updating the docs would be good.

That's a good point.  I hadn't realized it was exposed in the spec.

**Comment by Tianhao-Gu** [comment]
@MrCreosote @sean-mccorkle Thanks guys. Updated doc and added myself as module owner.

**Review by MrCreosote — APPROVED** [review]
Lgtm

**Pull request merged**